test/beast/zlib/deflate_stream.cpp

```diff
@@ -25,6 +25,138 @@ namespace zlib {
 
 class deflate_stream_test : public beast::unit_test::suite
 {
+    struct ICompressor {
```

Comment:
These are uuuuuugly but I guess that's a consequence of the zlib api...

Reply:
There's even a `const_cast` in there, because zlib wants its inputs to be non-const, although it doesn't modify them, while Beast rightfully requests a non-const version.
If any improvements come to your mind to make this slightly better, I'll happily apply them

Reply:
I want to believe that there is a more modern abstraction still waiting to be discovered which streamlines the inflate/deflate API and makes it easier to document and understand. Eventually I will investigate and hopefully find it. In the long term, the zlib part of Beast needs to be moved into its own repository and Boost library. In order to pass the Boost review though, some cleaning up will be needed (this doesn't have to happen now). Having it in a separate repository will make things easier on Travis for Beast and Zlib.